pyTelegramBotCaptcha/telebot_captcha.py

```diff
@@ -98,82 +127,105 @@ def text(self, your_text: str):
+        max_attempts: int = 2,
+        max_incorrect_to_auto_reload: int = 2,
+        add_noise: bool = True,
+        only_digits: bool = False,
+        custom_language: Optional[CustomLanguage] = None,
+    ) -> None:
         """
         Use this class to create a captcha options profile.
-        :param generator: The generator to use. Currently available: `"default"` and `"keyzend"`
+        :param generator: The generator to use. Currently available: `"default"` and `"keyzend"` and `"multicolor"` and `"math"`
         :param language: The language to use
         :param timeout: The timeout to use. (min: 30, max: 600)
         :param code_length: The length of the code (min: 4, max: 12)
         :param max_user_reloads: How many times can the user refresh his captcha. 0 or lower means he can not.
         :param max_attempts: How many attempts does the user have to solve the captcha. (min: 1)
-        :param max_incorrect_to_auto_reload: How many chars can be incorrect to auto reload. 
+        :param max_incorrect_to_auto_reload: How many chars can be incorrect to auto reload.
             if <= 0: on_not_correct event is triggered
             if >1: if n chars are incorrect the captcha is reloaded automatically (if an attempt is left)
         :param add_noise: Add noise to the image
         :param only_digits: Use only digits instead of hexdigits.
         :param custom_language: A custom language to use (overrides `language`)
 
-        NOTE: If `generator` is not set to `"default"`, some options will be ignored/overwritten 
+        NOTE: If `generator` is not set to `"default"`, some options will be ignored/overwritten
         """
-        if not generator.lower() in _generators: raise ValueError("This generator seems not to exist")
-        if not language.lower() in languages and not custom_language: raise NotImplementedError("This language is not suported yet")
-        if not MIN_TIMEOUT <= timeout <= MAX_TIMEOUT: raise ValueError("timeout must be between 30 and 600 (in seconds)")
-        if not MIN_CODE_LENGTH <= code_length <= MAX_CODE_LENGTH: raise ValueError("code_lenght must be between 4 and 12")
-        if max_attempts < 1: raise ValueError("max_attempts must be at least 1")
-
-        self._generator: str = generator.lower()
+        if not language.lower() in languages and not custom_language:
+            raise NotImplementedError("This language is not suported yet")
+        if not MIN_TIMEOUT <= timeout <= MAX_TIMEOUT:
+            raise ValueError("timeout must be between 30 and 600 (in seconds)")
+        if not MIN_CODE_LENGTH <= code_length <= MAX_CODE_LENGTH:
+            raise ValueError("code_lenght must be between 4 and 12")
+        if max_attempts < 1:
+            raise ValueError("max_attempts must be at least 1")
+
+        # no need to validate generator because EnumMeta will raise "ValueError: unknown is not a valid Generator"
+        self._generator: str = Generator(generator.lower()).name
```

Comment:
Hi @coder2020official . I understood from you that the commit to which Enum was added, does not accept strings as generator name, but I did not make any change in the structure of the code. It accepts strings as before, and the function of Enum is only verification as you see here